mcp/mcp-cli/src/main/java/software/amazon/smithy/java/mcp/cli/CliSystemPropertiesInitializer.java

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.java.mcp.cli;
+
+import java.util.Map;
+
+/**
+ * This class gives a hook to add system properties as the first step in main.
+ * Implementations should ensure they don't trigger any classloading in the implementations to avoid race between adding a system property and using it.
+ */
+public interface CliSystemPropertiesInitializer {
```

Comment:
Do we actually need this? Is it worth adding to our public interface?

Reply:
We can skip it but for Graal images there there is no other alternative than to create wrapper scripts and for some reason I can't find a gradle plugin that does it automatically.